doc/classes/ItemStackMetaData.adoc

```diff
@@ -0,0 +1,120 @@
+= ItemStackMetaData
+
+include::../include/config.adoc[]
+include::../include/types.adoc[]
+:description: Per-ItemStack persistent key-value store
+:keywords: meta, data, metadata, item, stack, storage, key, value, persistence
+
+ItemStackMetaData is a subclass of MetaData obtained via `stack:get_meta()`
+allowing for persistent storage of key-value pairs tied to ItemStacks.
+
+WARNING: ItemStack metadata is serialized with ItemStacks, increasing the ItemString length.
+Inventories have to store multiple ItemStrings, all of which an attacker will try to get to maximum length.
+Always limit the size of your ItemStackMetaData to keep inventories sendable.
+
+== Special Fields
+
+=== `description`
+
+The description to be shown when hovering over the item (see `ItemStack:get_description`).
+
+=== `short_description`
+
+A short description of the item (see `ItemStack:get_short_description`).
+
+=== `color`
+
+ColorString to use for hardware colorization of the stack.
+
+=== `palette_index`
+
+Palette index to use for hardware colorization of the stack (if the stack has a palette).
+
+=== `count_meta`
+
+String to show in inventory lists instead of the item count.
+
+=== `count_alignment`
+
+Integer (use with `:get_int` and `:set_int`).
```

Comment:
it should be documented which versions this is supported in. for example count_meta/alignment is only 5.6

Reply:
I disagree. We always target latest master.

Reply:
it should be documented, as these are client things, people need to be aware of what features they can use for which client versions, just like formspecs

Reply:
https://github.com/minetest/minetest_docs/pull/40/commits/d8f7b17f2f8e9777e72542b5f1dd6b703fc0591f